src/garage/envs/multi_task_env.py

```diff
@@ -0,0 +1,139 @@
+            raise ValueError('Invalid sample strategy.')
+
+        self._sample_strategy = sample_strategy
+
+        for task, env_cls in task_env_cls_dict.items():
+            env_kwargs = task_kwargs[task]
+            env_args = task_args[task]
+            task_env = env_cls(*env_args, **env_kwargs)
+            if self._task_envs:
+                if (task_env.observation_space.shape !=
+                        self.active_env.observation_space.shape):
+                    raise ValueError(
+                        'Observation space of all tasks should be same.')
+                if (task_env.action_space.shape !=
+                        self.active_env.action_space.shape):
+                    raise ValueError(
+                        'Action space of all tasks should be same.')
+
+            self._task_envs.append(task_env)
+
+    @property
+    def task_space(self):
+        """Task Space."""
+        one_hot_ub = np.ones(self.num_tasks)
+        one_hot_lb = np.zeros(self.num_tasks)
+        return akro.Box(one_hot_lb, one_hot_ub)
+
+    @property
+    def active_task(self):
+        """Id of currently active task."""
+        return self._active_task
+
+    @property
+    def active_env(self):
+        """Environment of currently active task."""
+        return self._task_envs[self.active_task]
+
+    @property
+    def num_tasks(self):
+        """Total number of tasks."""
+        return len(self._task_envs)
+
+    @property
+    def action_space(self):
+        """Task's Action space."""
+        return self.active_env.action_space
+
+    @property
+    def observation_space(self):
+        """Observation space."""
+        return self.active_env.observation_space
+
+    @property
+    def horizon(self):
+        """Horizon of currently active task env."""
+        return self.active_env.horizon
+
+    @property
+    def valid_sample_strategies(self):
+        """Supported sample strategies."""
+        return self._valid_sample_strategies
+
+    def reset(self, **kwargs):
+        """Reset the active task env."""
+        return self.active_env.reset(**kwargs)
+
+    def sample_tasks(self, n_tasks=None):
+        """
+        Sample tasks according to self.sample_strategy.
+
+        Args:
+            n_tasks (int): Number of tasks to sample. Defaults to
+            self.num_tasks if not provided.
+        """
+        if not n_tasks:
+            n_tasks = self.num_tasks
+
+        if self._sample_strategy == 'all':
+            tasks = list(range(n_tasks))
+        elif self._sample_strategy == 'random':
+            tasks = np.random.randint(0, self.num_tasks, size=n_tasks).tolist()
+
+        return tasks
+
+    def set_task(self, task):
+        """Set the active task."""
+        self._active_task = task % len(self._task_envs)
+
+    def step(self, action):
+        """gym.Env step for the active task env."""
+        obs, reward, done, info = self.active_env.step(action)
```

Comment:
I think extending `info` with task information would be super helpful. For example, a task name. Task onehot can also be helpful too.